src/kOS.Safe/Compilation/Opcode.cs

```diff
@@ -1187,14 +1189,23 @@ public override void Execute(ICpu cpu)
                     functionPointer = cpu.GetValue(functionName);
                 }
             }
- 
-            if (functionPointer is int) 
+            IUserDelegate userDelegate = functionPointer as IUserDelegate;
+            if (userDelegate != null)
+                functionPointer = userDelegate.EntryPoint;
+
+            if (functionPointer is int)
             {
                 ReverseStackArgs(cpu);
                 int currentPointer = cpu.InstructionPointer;
                 DeltaInstructionPointer = (int)functionPointer - currentPointer;
                 var contextRecord = new SubroutineContext(currentPointer+1);
                 cpu.PushAboveStack(contextRecord);
+                if (userDelegate != null)
+                {
+                    // Reverse-push the closure's scope record, just after the function return context got put on the stack.
+                    for (int i = userDelegate.Closure.Count - 1 ; i >= 0 ; --i)
+                        cpu.PushAboveStack(userDelegate.Closure[i]);
+                }
```

Comment:
In general I'm starting to get a bit nervous about how long the body of OpcodeCall.Execute() is getting.  Eventually it should be split up into separate sections *somehow*, but every time I look into how I'd do it I just shudder and leave it as-is.  The reason it's all intertwined is that I don't want to cut-n-paste as much as I'd have to of I just had separate cases splitting into their own subroutines (if it's direct, do this, if it's indirect with a C# delegate do this, if it's indirect with a UserDelegate do this, if it's indirect with an integer delegate do this...).  These cases end up being about 75% the same as each other.

But what makes it hard to split up that other 25% into separate routines is that it's not the *same* 25%.  The bits they share in common are not the SAME - so you get an overlapping Venn diagram of which cases call which code, that isn't nice and clear to separate.

So, I'm saying I know the current long length with intermingled conditions is poor form, but fixing it is nontrivial and should probably be its own refactor rather than doing it simultaneously with a PR that is changing functionality.

When it gets refactored, it should be the sort of frozen-functionality PR that allows for regression testing (make sure this didn't change anything....).